File: python/contest-2-mar/2.py
from typing import List
import heapq
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        def create_min_heap(arr):
            min_heap = []
            for num in arr:
                heapq.heappush(min_heap, num)
            return min_heap
        min_heap = create_min_heap(nums)
        result = 0
        while True:
            a = heapq.heappop(min_heap)
            if(a >= k):
                return result
            b = heapq.heappop(min_heap)
            result+=1
            heapq.heappush(min_heap, min(a,b) * 2 + max(a,b))

a = Solution()
print(a.minOperations([1,1,2,4,9],20))

File: contests/429/2.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var maxDistinctElements = function (nums, k) {
  nums.sort((a, b) => a - b)
  let max = nums[nums.length - 1] + k + 1
  for (let i = nums.length - 1; i >= 0; i--) {
    nums[i] = Math.max(Math.min(max - 1, nums[i] + k), nums[i] - k)
    max = nums[i]
  }

  const set = new Set()
  for (const num of nums) {
    set.add(num)
  }

  return set.size
};
console.log(maxDistinctElements([1, 2, 2, 3, 3, 4], 2))